control-plane/rest/src/versions/v0.rs

```diff
@@ -176,6 +177,37 @@ pub struct GetBlockDeviceQueryParams {
     pub all: Option<bool>,
 }
 
+/// The difference types of watches
+#[derive(Deserialize, Apiv2Schema)]
+#[serde(rename_all = "camelCase")]
+pub struct WatchTypeQueryParam {
+    /// URL callback
+    pub callback: RestUri,
+}
+
+/// Watch Resource in the store
+#[derive(Serialize, Deserialize, Debug, Clone, Apiv2Schema, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct RestWatch {
+    /// id of the resource to watch on
+    pub resource: String,
+    /// callback used to notify the watcher of a change
+    pub callback: String,
+}
+
+impl TryFrom<&Watch> for RestWatch {
+    type Error = ();
+    fn try_from(value: &Watch) -> Result<Self, Self::Error> {
+        match &value.callback {
+            WatchCallback::Uri(uri) => Ok(Self {
+                resource: value.resource.to_string(),
+                callback: uri.to_string(),
+            }),
+            // _ => Err(()),
```

Comment:
Does this need to be uncommented?

Reply:
No, because we only have one we can't other `_` for others. I could just deleted, just wanted to make it clear that everything else is not implemented yet, maybe I'll add such comment

Reply:
A comment sounds good.